File: .agent/scripts/lint.py
import subprocess
import argparse
import os
import sys
import re

def remove_ansi_codes(text):
    # Regex to remove ANSI escape sequences
    ansi_escape = re.compile(r'\x1B(?:[@-Z\\-_]|\[[0-?]*[ -/]*[@-~])')
    return ansi_escape.sub('', text)

def run_lint(file_path):
    log_file = f"{file_path}.log"
    results = []

    results.append(sys.executable + ' ' + ' '.join(sys.argv) + "\n" )
    
    def run_tool(cmd_list, name):
        results.append(f"=== {name} ===\n")
        try:
            is_windows = os.name == 'nt'
            # Use subprocess.list2cmdline only for shell=True on Windows
            cmd = subprocess.list2cmdline(cmd_list) if is_windows else cmd_list
            
            process = subprocess.run(
                cmd,
                capture_output=True,
                text=True,
                encoding='utf-8',
                errors='replace',
                shell=is_windows
            )
            
            stdout = remove_ansi_codes(process.stdout or "")
            stderr = remove_ansi_codes(process.stderr or "")
            
            results.append(f"Exit Code: {process.returncode}\n")
            if stdout.strip():
                results.append("STDOUT:\n" + stdout + "\n")
            if stderr.strip():
                results.append("STDERR:\n" + stderr + "\n")
                
        except Exception as e:
            results.append(f"Execution Error: {str(e)}\n")
        results.append("\n" + "="*40 + "\n\n")

    # 1. Autocorrect
    # Removed --no-color as it was not supported by the local version
    # Our remove_ansi_codes will clean up the output instead
    run_tool(["autocorrect", "--lint", file_path, "--format", "diff"], "Autocorrect Linting")

    # 2. Markdownlint-cli2
    run_tool(["markdownlint-cli2", file_path], "Markdownlint-cli2")

    try:
        with open(log_file, "w", encoding="utf-8") as f:
            f.writelines(results)
        print(f"Linting complete. Results saved to: {log_file}")
    except Exception as e:
        print(f"Failed to write log file: {e}", file=sys.stderr)
        return 1

    return 0

if __name__ == "__main__":
    parser = argparse.ArgumentParser(description="Lint a markdown file and save results to a log.")
    parser.add_argument("--file", required=True, help="Path to the markdown file to lint.")
    
    args = parser.parse_args()
    target_file = os.path.abspath(args.file)
    
    if not os.path.exists(target_file):
        print(f"Error: File '{target_file}' does not exist.", file=sys.stderr)
        sys.exit(1)
        
    # Always exit with 0 to prevent CI/CD failures on lint errors
    run_lint(target_file)
    sys.exit(0)

File: scripts/skills/review/lint.js
const { spawnSync } = require('child_process');
const fs = require('fs');
const path = require('path');

function removeAnsiCodes(text) {
    // Regex to remove ANSI escape sequences
    const ansiEscape = /[\u001b\u009b][[()#;?]*(?:[0-9]{1,4}(?:;[0-9]{0,4})*)?[0-9A-ORZcf-nqry=><]/g;
    return text.replace(ansiEscape, '');
}

function runLint(filePath) {
    const rootProcess = spawnSync('git', ['rev-parse', '--show-toplevel'], { 
        encoding: 'utf8',
        shell: process.platform === 'win32'
    });
    const rootDir = rootProcess.stdout.trim();
    if (!rootDir) {
        console.error("无法找到 Git 根目录。");
        return 1;
    }

    const logFile = `${filePath}.log`;
    let results = [];

    results.push(`node ${process.argv.join(' ')}\n`);

    function runTool(cmd, args, name) {
        results.push(`=== ${name} ===\n`);
        try {
            const processRes = spawnSync(cmd, args, {
                encoding: 'utf8',
                cwd: rootDir,
                shell: process.platform === 'win32'
            });

            const stdout = removeAnsiCodes(processRes.stdout || "");
            const stderr = removeAnsiCodes(processRes.stderr || "");

            results.push(`Exit Code: ${processRes.status}\n`);
            if (stdout.trim()) {
                results.push("STDOUT:\n" + stdout + "\n");
            }
            if (stderr.trim()) {
                results.push("STDERR:\n" + stderr + "\n");
            }

        } catch (e) {
            results.push(`Execution Error: ${e.message}\n`);
        }
        results.push("\n" + "=".repeat(40) + "\n\n");
    }

    // 1. Autocorrect
    runTool("autocorrect", ["--lint", filePath, "--format", "diff"], "Autocorrect Linting");

    // 2. Markdownlint-cli2
    runTool("markdownlint-cli2", [filePath], "Markdownlint-cli2");

    try {
        fs.writeFileSync(logFile, results.join(''), 'utf8');
        console.log(`Linting complete. Results saved to: ${logFile}`);
    } catch (e) {
        console.error(`Failed to write log file: ${e.message}`);
        return 1;
    }

    return 0;
}

const args = process.argv.slice(2);
const fileArgIndex = args.indexOf('--file');
if (fileArgIndex === -1 || fileArgIndex + 1 >= args.length) {
    console.error("Usage: node lint.js --file <path-to-markdown-file>");
    process.exit(1);
}

const targetFile = path.resolve(args[fileArgIndex + 1]);

if (!fs.existsSync(targetFile)) {
    console.error(`Error: File '${targetFile}' does not exist.`);
    process.exit(1);
}

// Always exit with 0 to prevent CI/CD failures on lint errors
runLint(targetFile);
process.exit(0);
